sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/SpannerWriteIT.java

```diff
@@ -128,19 +140,72 @@ public void testWrite() throws Exception {
                 .withInstanceId(options.getInstanceId())
                 .withDatabaseId(databaseName));
 
-    p.run();
-    DatabaseClient databaseClient =
-        spanner.getDatabaseClient(
-            DatabaseId.of(
-                project, options.getInstanceId(), databaseName));
+    PipelineResult result = p.run();
+    result.waitUntilFinish();
+    assertThat(result.getState(), is(PipelineResult.State.DONE));
+    assertThat(countNumberOfRecords(), equalTo((long) numRecords));
```

Comment:
Hm I wonder, can you use SpannerIO.read() here instead? (or readAll(), gated by Wait.on(write)) - that would be another cool demonstration of Wait.

Reply:
Isn't it better for the test to compute that without going through beam? `testSequentialWrite` already tests `Wait.on`.

Reply:
In general, "write-then-read" tests are preferred in Beam, where both write and read are done using your transform. They can be done in separate pipelines in one test method, or in one pipeline. I'm OK keeping this as-is for this PR though.